modules/free/arrow-free-data/src/test/kotlin/arrow/free/CofreeTest.kt

```diff
@@ -32,19 +32,23 @@ import arrow.test.UnitSpec
 import arrow.test.concurrency.SideEffect
 import arrow.test.laws.ComonadLaws
 import arrow.typeclasses.Eq
+import io.kotlintest.properties.Gen
 import io.kotlintest.shouldBe
 
 class CofreeTest : UnitSpec() {
 
   init {
 
-    testLaws(ComonadLaws.laws(Cofree.comonad(), {
+    val cf: (Int) -> Cofree<ForOption, Int> = {
       val sideEffect = SideEffect()
       unfold(Option.functor(), sideEffect.counter) {
         sideEffect.increment()
         if (it % 2 == 0) None else Some(it + 1)
       }
-    }, Eq { a, b ->
+    }
+    val g = Gen.int().map(cf) as Gen<Kind<Kind<ForCofree, ForOption>, Int>>
```

Comment:
You can make this a nicer gen by generating a non-empty list and folding that into cofree. 
`Cofree` in haskell is represented as `a <: F<Cofree<F, A>>` which with `a <: Option<Cofree<ForOption, A>>` is exactly the same as a non-empty list.

Reply:
like this? https://github.com/arrow-kt/arrow/blob/master/modules/recursion/arrow-recursion-data/src/test/kotlin/arrow/recursion/CofreeBirecursiveTest.kt#L66

Reply:
Damn I forgot I wrote something like this already :smile: 
Yes same concept, I am using it there as a co-algebra for recursion schemes, but this can also be written as a simple unfold (which is technically the same thing)